h2o-automl/src/main/java/ai/h2o/automl/modeling/StackedEnsembleStepsProvider.java

```diff
@@ -130,6 +139,69 @@ protected boolean isStackedEnsemble(Key<Model> key) {
+                                    model._parms, "_monotone_constraints",
+                                    PojoUtils.FieldNaming.CONSISTENT);
+                            return mc != null && mc.length > 0;
+                        } catch (IllegalArgumentException e) {
+                            return false;
+                        }
+                    }
+
+                    @Override
+                    protected boolean canRun() {
+                        boolean canRun = super.canRun();
+                        if (!canRun) return false;
+                        int monotoneModels=0;
+                        for (Key<Model> modelKey: getTrainedModelsKeys()) {
+                            if (hasMonotoneConstrains(modelKey))
+                                monotoneModels++;
+                            if (monotoneModels >= 2)
+                                return true;
+                        }
+                        if (monotoneModels == 1) {
+                            aml().job().update(getAllocatedWork().consume(),
+                                    "Only one monotone base model; skipping this StackedEnsemble");
+                            aml().eventLog().info(EventLogEntry.Stage.ModelTraining,
+                                    String.format("Skipping StackedEnsemble '%s' since there is only one monotone model to stack", _id));
+                        } else {
+                            aml().job().update(getAllocatedWork().consume(),
+                                    "No monotone base model; skipping this StackedEnsemble");
+                            aml().eventLog().info(EventLogEntry.Stage.ModelTraining,
+                                    String.format("Skipping StackedEnsemble '%s' since there is no monotone model to stack", _id));
+                        }
+                        return false;
+                    }
+
+                    @Override
+                    @SuppressWarnings("unchecked")
+                    protected Key<Model>[] getBaseModels() {
+                        return Stream.of(getTrainedModelsKeys())
+                                .filter(k -> !isStackedEnsemble(k) && hasMonotoneConstrains(k))
+                                .toArray(Key[]::new);
+                    }
+
+                    @Override
+                    protected StackedEnsembleParameters getStackedEnsembleParameters(Key<Model>[] baseModels, boolean isLast) {
+                        StackedEnsembleParameters params = super.getStackedEnsembleParameters(baseModels, isLast);
+                        GLMModel.GLMParameters metalearnerParams = new GLMModel.GLMParameters();
+                        // AUTO has non negative constraint to make it easier to reason about the whole ensemble
+                        params._metalearner_algorithm = Metalearner.Algorithm.AUTO;
+                        params._metalearner_parameters = metalearnerParams;
+                        return params;
```

Comment:
can still be improved.

in your `getStackedEnsembleParameters`, there's a call to `stackedEnsembleParameters.initMetalearnerParams();`.
This last method is overloaded so that you can directly pass the metalearner algo to initialize the parameters.
Note that with your current logic, the metalearner params are created twice: 
1. once in `super.getStackedEnsembleParameters` where some additional generic params about CV are also set.
2. once here, completely erasing the previous ones, including the k`keep_cv.__` params.
It's probably not your objective.

If you just want to enforce metalearner algo, (btw, it think it's doable only using the non-official `algo_parameters` functionality), you should be able to simply call sth like:
```
return super.getStackedEnsembleParameters(Metalearner.Algorithm.AUTO, baseModels, isLast);
```
Also note that if one day we need to enfore GLM, you can still do
```
StackedEnsembleParameters params = super.getStackedEnsembleParameters(Metalearner.Algorithm.GLM, baseModels, isLast);
((GLMParameters)(params._metalearner_parameters))._non_negative = true
```
still without erasing some of the params set in `getStackedEnsembleParameters`.

Reply:
Thanks @sebhrusen! 
Do I understand correctly, that I don't need to implement `getStackedEnsembleParameters` at all? 
It will work well unless we change the implementation of `AUTO` metalearner or if we allow to specify default metalearner in AutoML or if the user uses the non-official `algo_parameters` etc...

Reply:
I'm not fully sure we should remove the `getStackedEnsembleParameters`.
Intuitively, I would have kept it and just enforce the `AUTO` metalearner, but after checking, I realize that the `algo_parameters` are applied after this anyway (in `trainModel`) so, it would give actually the false impression that it's enforcing something...

I just want to keep code as simple as possible, without leaving any blind spot, so I just created those tickets:
https://0xdata.atlassian.net/browse/PUBDEV-7570  with a note so that it get fixed during next improvement of `algo_parameters`.
https://0xdata.atlassian.net/browse/PUBDEV-7571 so that we keep in mind the specificity of the monotonic SE.